bin/knc.h

```diff
@@ -59,11 +59,11 @@ extern prefs_t prefs;
 typedef struct write_buffer_s {
 	char			in[GSTD_MAXPACKETCONTENTS];
 	char			in_valid;
-	size_t			in_len;
+	ssize_t			in_len;
```

Comment:
This requires thinking.  I think these really should be `size_t` -- unsigned -- with negative `ssize_t` return values handled where they are produced.

Reply:
Per-discussion, you went for this change to keep it tiny, but I'd rather we keep these as unsigned.